### PR TITLE
HttpClient now is able to do HTTP PUT in addition to POST and GET.

### PR DESCRIPTION
**Pull request by deathbaba**

### CODE REVIEW

**Comment by deathbaba** [comment]
@mzhurovich Дима сказал заассайнить тебя. Глянешь на реквест, пожалуйста?

**Comment by mzhurovich** [comment]
Ок, сегодня постараюсь глянуть.



10 апр. 2015 г., в 6:56, Alex Zolotarev <notifications@github.com>
написал(а):

@mzhurovich <https://github.com/mzhurovich> Дима сказал заассайнить тебя.
Глянешь на реквест, пожалуйста?

—
Reply to this email directly or view it on GitHub
<https://github.com/KnowSheet/Alohalytics/pull/33#issuecomment-91420228>.

File: src/android/java/org/alohalytics/HttpTransport.java [hunk 1]
```diff
@@ -93,14 +97,18 @@ public static Params run(final Params p) throws IOException, NullPointerExceptio
...
+            Log.d(TAG, "Sent " + p.httpMethod + " with file of size " + file.length());
         }
       }
-      // GET data from the server or receive POST response body
+      // GET data from the server or receive response body
       p.httpResponseCode = connection.getResponseCode();
       if (p.debugMode)
         Log.d(TAG, "Received HTTP " + p.httpResponseCode + " from server.");
-      p.receivedUrl = connection.getURL().toString();
+      if (p.httpResponseCode < 399 && p.httpResponseCode >= 300) {
```
Comment (mzhurovich) [inline src/android/java/org/alohalytics/HttpTransport.java]: `(x >= a && x < b)` would be easier to read

File: src/android/java/org/alohalytics/HttpTransport.java [hunk 2]
```diff
@@ -150,14 +158,13 @@ public static Params run(final Params p) throws IOException, NullPointerExceptio
     public String url = null;
     // Can be different from url in case of redirects.
     public String receivedUrl = null;
-    // SHOULD be specified for any POST request (any request where we send data to the server).
+    public String httpMethod = null;
+    // SHOULD be specified for any POST-like request (any request where we send data to the server).
     // On return, contains received Content-Type or null.
     public String contentType = null;
-    // Can be specified for any POST request (any request where we send data to the server).
+    // Can be specified for any POST-like request (any request where we send data to the server).
```
Comment (mzhurovich) [inline src/android/java/org/alohalytics/HttpTransport.java]: 'POST-like' sounds a bit confusing.
Maybe something like 'Can be specified for any request whose method allows non-empty body.'

**Comment by mzhurovich** [comment]
@deathbaba Looks good, but I'm not very familiar with this code base.
One small suggestion: when setting the request body, I would add a check, if the request method is appropriate. It could be useful to avoid potential errors in the future.

**Comment by deathbaba** [comment]
Thanks! Can you please also check 9938ca0 (and merge request if everything is OK) ?

**Pull request merged**